zone-setup/src/bin/zone-setup.rs

```diff
@@ -22,12 +25,279 @@ use uzers::{get_group_by_name, get_user_by_name};
+                .args(["-P", "''", &self.user])
+                .output()
+                .map_err(|err| {
+                    CmdError::Failure(anyhow!(
+                        "Could not execute `usermod -P '' {}`: {}",
+                        self.user,
+                        err
+                    ))
+                })?;
+
+            if !cmd.status.success() {
+                return Err(CmdError::Failure(anyhow!(
+                    "Could not remove user profiles: {} status: {}",
+                    self.user,
+                    cmd.status
+                )));
+            }
+            // println!("usermod -P '' {}", self.user)
+        };
+
+        if let Some(homedir) = self.homedir {
+            info!(&log, "Set up home directory and startup files"; "user" => self.user.clone(), "home directory" => homedir.clone());
+            create_dir_all(&homedir).map_err(|err| {
+                CmdError::Failure(anyhow!(
+                    "Could not execute create directory {} and its parents: {}",
+                    &homedir,
+                    err
+                ))
+            })?;
+
+            let mut home_bashrc = homedir.clone();
+            home_bashrc.push_str("/.bashrc");
+            copy("/root/.bashrc", &home_bashrc).map_err(|err| {
+                CmdError::Failure(anyhow!(
+                    "Could not copy file from /root/.bashrc to {}: {}",
+                    &homedir,
+                    err
+                ))
+            })?;
+
+            let mut home_profile = homedir.clone();
+            home_profile.push_str("/.profile");
+            copy("/root/.profile", &home_profile).map_err(|err| {
+                CmdError::Failure(anyhow!(
+                    "Could not copy file from /root/.profile to {}: {}",
+                    &homedir,
+                    err
+                ))
+            })?;
+
+            // Not using std::os::unix::fs::chown here because it doesn't support
+            // recursive option.
+            let cmd = std::process::Command::new("chown")
+                .args(["-R", &self.user, &homedir])
+                .output()
+                .map_err(|err| {
+                    CmdError::Failure(anyhow!(
+                        "Could not execute `chown -R {} {}`: {}",
+                        self.user,
+                        homedir,
+                        err
+                    ))
+                })?;
+
+            if !cmd.status.success() {
+                return Err(CmdError::Failure(anyhow!(
+                    "Could not change ownership: {} status: {}",
+                    homedir,
+                    cmd.status
+                )));
+            }
+
+            // println!("mkdir -p {}", homedir);
+            // println!("cp /root/.bashrc {}/.bashrc", homedir);
+            // println!("cp /root/.profile {}/.profile", homedir);
+            // println!("chown -R {} {}", self.user, homedir);
+        }
+        Ok(())
+    }
+}
```

Comment:
@citrus-it, still a bit of cleanup to do, but I'd love to hear your opinion on this approach! Happy to modify/toss away if this is overkill.